### DISCUSSION TITLE
issue with glob matching at the last part of path

### DISCUSSION THREAD

**Discussion by dealy663** [babashka/fs#66] [Q&A]
I've run into a problem when trying to use fs/glob to match a path which matches the end of the pattern. For instance
(fs/glob "/Users/derek/tmp/file1" "file1") => [] or
(fs/glob "/Users/derek/tmp/file1" "**file1") => []  both return nothing

but pointing to its parent dir works
(fs/glob "/Users/derek/tmp" "file1") => [#object[sun.nio.fs.UnixPath 0x62a73d2b "/Users/derek/tmp/file1"]]

The above is counter-intuitive to me. Also I can match in this scenario using the standard Java FileSystems matcher and get the expected behavior:
(def jfs (FileSystems/getDefault))
(def match3 (.getPathMatcher jfs "glob:**file1"))
(def f1 (io/file "/Users/derek/tmp/file1"))
(.matches match3 (fs/canonicalize f1)) => _true_

Maybe I'm misunderstanding something about how you intended glob to be used, but this sounds like a bug to me.

Thanks for taking a look at this

Derek

**Comment by borkdude** [comment]
`fs/glob` by default doesn't match on the root argument. The way it's intended to be used is:

```
(fs/glob root-directory pattern)
```

**Comment by borkdude** [comment]
Compare with Python:

```
mkdir test
cd test
mkdir foo
python3
>>> import glob
>>> glob.glob('*')
['foo']
```

I.e. it doesn't return the "current" directory in which it searches.
